feat!: slimmer wallet interface

Thunkar opened the pull request:
Keep removing methods that do not belong in the wallet interface. This time:

- `createPublicAuthwit`: This is just a contract interaction. A few convenience methods have been added, and the `TestWallet` keeps the interface to avoid pain the e2e tests, but from now on public authwits are just that: glorified ContractInteractions
- `getPublicEvents`: Nuked also from PXE. There's 0 private information required to decode them, moved to an utility instead

`createAuthwit` in the Wallet loses the ability to compute them from `ContractFunctionInteraction`s since they're not serializable (still kept it in `TestWallet` for the same reasons as above)

Added a new `getChainInfo` method that returns chainId and version and can be used to generate authwits externally and generally to gather information about the chain the wallet is connected to.

Also added a proper validated interface to the `Wallet` that will be used soon ™️

benesjan commented on playground/src/wallet/extension_wallet.ts:
```diff
@@ -0,0 +1,58 @@
+
+type FunctionsOf<T> = { [K in keyof T as T[K] extends Function ? K : never]: T[K] };
+
+export class ExtensionWallet {
```
Ser, what is an extension wallet? Docs pls 😝

Thunkar replied:
whoops, that's actually an extra file from my experiments

benesjan reviewed (approved):
Looks good! LGTM

Just couldn't help and not bother you with docs. Hehe

Thunkar committed:
feat!: slimmer wallet interface

Keep removing methods that do not belong in the wallet interface. This time:

- `createPublicAuthwit`: This is just a contract interaction. A few convenience methods have been added, and the `TestWallet` keeps the interface to avoid pain the e2e tests, but from now on public authwits are just that: glorified ContractInteractions
- `getPublicEvents`: Nuked also from PXE. There's 0 private information required to decode them, moved to an utility instead

`createAuthwit` in the Wallet loses the ability to compute them from `ContractFunctionInteraction`s since they're not serializable (still kept it in `TestWallet` for the same reasons as above)

Added a new `getChainInfo` method that returns chainId and version and can be used to generate authwits externally and generally to gather information about the chain the wallet is connected to.

Also added a proper validated interface to the `Wallet` that will be used soon ™️

Co-authored-by: thunkar <gregojquiros@gmail.com>
